tools/deploy_metrics_and_monitoring_resources.sh

```diff
@@ -0,0 +1,155 @@
+    $SED '' "s/namespace: .*/namespace: $NAMESPACE/" $DIR/metrics/prometheus-operator.yaml > $DIR/metrics/prometheus-operator-deploy.yaml
+fi
+kubectl apply -f $DIR/metrics/prometheus-operator-deploy.yaml -n $NAMESPACE
+
+echo "Waiting for Prometheus operator to be ready..."
+kubectl rollout status deployment/strimzi-prometheus-operator -w -n $NAMESPACE
+echo "...Prometheus operator ready"
+
+rm $DIR/metrics/prometheus-operator-deploy.yaml
+
+kubectl create secret generic additional-scrape-configs --from-file=$DIR/metrics/examples/prometheus/additional-properties/prometheus-additional.yaml
+kubectl create secret generic alertmanager-alertmanager --from-file=alertmanager.yaml=$DIR/metrics/examples/prometheus/alertmanager-config/alert-manager-config.yaml
+
+
+if [[ "$OSTYPE" == "linux-gnu" ]]; then
+    $SED "s/namespace: .*/namespace: $NAMESPACE/" $DIR/metrics/examples/prometheus/install/prometheus.yaml > $DIR/metrics/examples/prometheus/install/prometheus-deploy.yaml
+elif [[ "$OSTYPE" == "darwin"* ]]; then
+    $SED '' "s/namespace: .*/namespace: $NAMESPACE/" $DIR/metrics/examples/prometheus/install/prometheus.yaml > $DIR/metrics/examples/prometheus/install/prometheus-deploy.yaml
+fi
+
+kubectl apply -f $DIR/metrics/examples/prometheus/install/prometheus-rules.yaml -n $NAMESPACE
+kubectl apply -f $DIR/metrics/examples/prometheus/install/prometheus-deploy.yaml -n $NAMESPACE
+rm $DIR/metrics/examples/prometheus/install/prometheus-deploy.yaml
+kubectl apply -f $DIR/metrics/examples/prometheus/install/strimzi-service-monitor.yaml -n $NAMESPACE
+kubectl apply -f $DIR/metrics/examples/prometheus/install/alert-manager.yaml -n $NAMESPACE
+
+# Delay for allowing Prometheus operator to create the Alert manager statefulset
+sleep 15
+
+amReplicas=$(kubectl get statefulsets alertmanager-alertmanager -o jsonpath="{.spec.replicas}" -n $NAMESPACE)
+echo "Waiting for Alert manager to be ready..."
+readyReplicas="0"
+while [ "$readyReplicas" != "$amReplicas" ]
+do
+    sleep 2
+    readyReplicas=$(kubectl get statefulsets alertmanager-alertmanager -o jsonpath="{.status.readyReplicas}" -n $NAMESPACE)
+done
+echo "...Alert manager ready"
+
+# Delay for allowing Prometheus operator to create the Prometheus statefulset
+sleep 15
+
+pReplicas=$(kubectl get statefulsets prometheus-prometheus -o jsonpath="{.spec.replicas}" -n $NAMESPACE)
+echo "Waiting for Prometheus to be ready..."
+readyReplicas="0"
+while [ "$readyReplicas" != "$pReplicas" ]
+do
+    sleep 2
+    readyReplicas=$(kubectl get statefulsets prometheus-prometheus -o jsonpath="{.status.readyReplicas}" -n $NAMESPACE)
+done
+echo "...Prometheus ready"
+
+rm $DIR/metrics/prometheus-operator.yaml
+
+# Grafana
+kubectl apply -f $DIR/metrics/examples/grafana/grafana.yaml -n $NAMESPACE
+
+echo "Waiting for Grafana server to be ready..."
+kubectl rollout status deployment/grafana -w -n $NAMESPACE
+echo "...Grafana server ready"
+
+kubectl expose service/grafana -n $NAMESPACE
```

Comment:
You should not do this since this is possibly a security risk. You expose unsecured service without the user possibly knowing. You should do that only when a parameter is set or not at all.